src/test/java/jewellery/inventory/unit/service/OrganizationServiceTest.java

```diff
@@ -97,11 +102,103 @@ void createOrganizationSuccessfully() {
     assertEquals(actual, organizationResponseDto);
   }
 
+  @Test
+  void getUsersInOrganizationSuccessfully() {
+    when(organizationRepository.findById(organizationWithUser.getId()))
+        .thenReturn(Optional.of(organizationWithUser));
+    when(authService.getCurrentUser()).thenReturn(executorResponseDto);
+    when(userService.getUser(any(UUID.class))).thenReturn(user);
+    when(organizationMapper.toUserInOrganizationResponseDtoResponse(organizationWithUser))
+        .thenReturn(new ArrayList<>());
+
+    List<UserInOrganizationResponseDto> actual =
+        organizationService.getAllUsersInOrganization(organizationWithUser.getId());
+
+    assertNotNull(actual);
+    assertEquals(actual.size(), 0);
+  }
+
+  @Test
+  void updateUserPermissionsInOrganizationSuccessfully() {
+    when(organizationRepository.findById(organizationWithUser.getId()))
+        .thenReturn(Optional.of(organizationWithUser));
+    when(authService.getCurrentUser()).thenReturn(executorResponseDto);
+    when(userService.getUser(any(UUID.class))).thenReturn(user);
+    when(organizationMapper.toResponse(organizationWithUser)).thenReturn(organizationResponseDto);
+
+    OrganizationResponseDto actual =
+        organizationService.updateUserPermissionsInOrganization(
+            organizationWithUser.getId(),
+            user.getId(),
+            List.of(OrganizationPermission.DESTROY_ORGANIZATION));
+
+    assertNotNull(actual);
+    assertEquals(actual, organizationResponseDto);
+  }
+
+  @Test
+  void deleteUserInOrganizationSuccessfully() {
+    when(authService.getCurrentUser()).thenReturn(executorResponseDto);
+    when(userService.getUser(any(UUID.class))).thenReturn(user);
+    when(organizationRepository.save(organizationWithUser)).thenReturn(organizationWithUser);
+    when(organizationRepository.findById(organizationWithUser.getId()))
+        .thenReturn(Optional.of(organizationWithUser));
+
+    when(organizationMapper.toResponse(organizationWithUser)).thenReturn(organizationResponseDto);
+
+    OrganizationResponseDto actual =
+        organizationService.deleteUserInOrganization(user.getId(), organizationWithUser.getId());
+    assertNotNull(actual);
+    assertEquals(actual, organizationResponseDto);
+  }
+
   @Test
   void testGetOrganizationShouldThrowWhenItsNotFound() {
     UUID fakeId = UUID.fromString("58bda8d1-3b3d-4319-922b-f5bb66623d71");
     assertThrows(
         OrganizationNotFoundException.class,
         () -> organizationService.getOrganizationResponse(fakeId));
   }
+
+  @Test
+  void deleteUserInOrganizationThrowsExceptionWhenNoManageUsersPermission() {
+    when(authService.getCurrentUser()).thenReturn(executorResponseDto);
+    when(userService.getUser(any(UUID.class))).thenReturn(new User());
```

Comment:
We can be a bit more specific here to let the reader know that these are connected.
```suggestion
    when(userService.getUser(executorResponseDto.getId())).thenReturn(new User());
```